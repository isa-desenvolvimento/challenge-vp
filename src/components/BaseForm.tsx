import { Formik, Form } from 'formik'

// eslint-disable-next-line @typescript-eslint/no-explicit-any
export const BaseForm = ({ initialValues, baseForm, onSubmit }: any) => {
  const renderForm = () => {
    return (
      <Formik
        onSubmit={(data, action) => onSubmit(data, action)}
        initialValues={initialValues}
      >
        {() => <Form>{baseForm}</Form>}
      </Formik>
    )
  }

  return <>{renderForm()}</>
}
