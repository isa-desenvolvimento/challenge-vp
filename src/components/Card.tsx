import {
  StyleCardContainer,
  StyleCardContent,
  StyleCardFooter,
  StyleCardHeader,
} from '@/styles/Card'
import { CardProps } from '@/types/global'

export const Card = ({
  header,
  content,
  footer,
  primary,
  height,
}: CardProps) => {
  const renderHeader = () =>
    header && <StyleCardHeader primary={primary}>{header}</StyleCardHeader>
  const renderContent = () =>
    content && (
      <StyleCardContent primary={primary} height={height}>
        {content}
      </StyleCardContent>
    )
  const renderFooter = () =>
    footer && <StyleCardFooter primary={primary}>{footer}</StyleCardFooter>

  return (
    <StyleCardContainer role="card">
      {renderHeader()}
      {renderContent()}
      {renderFooter()}
    </StyleCardContainer>
  )
}
