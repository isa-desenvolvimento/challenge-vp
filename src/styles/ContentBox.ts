import styled from 'styled-components'

export const StyleContainer = styled.main`
  grid-area: contentBox;
  overflow: scroll;
  width: 100%;
  height: 100%;
  background: ${({ theme }) => theme.colors.background};
`
