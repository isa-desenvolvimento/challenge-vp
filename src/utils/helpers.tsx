import { render } from '@testing-library/react'
import GlobalContext from '../context'

export const renderWithProvider = (children: React.ReactNode) => {
  render(<GlobalContext>{children}</GlobalContext>)
}
