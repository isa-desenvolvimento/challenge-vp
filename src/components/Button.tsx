import { Icon } from '@/components/Icon'
import { StyleButton } from '@/styles/Button'
import { ButtonProps } from '@/types/global'

export const Button = ({
  iconProps,
  primary,
  onClick,
  children,
  size,
  type,
}: ButtonProps) => {
  const renderIcon = () => iconProps && <Icon faIcon={iconProps}></Icon>

  return (
    <StyleButton
      role="button"
      arial-label="button"
      primary={primary}
      size={size}
      type={type}
      onClick={onClick}
    >
      {renderIcon()}
      {children}
    </StyleButton>
  )
}
