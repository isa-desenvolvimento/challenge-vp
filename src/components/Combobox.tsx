import { ComboboxProps } from '@/types/global'

import { DropdownWrapper, StyledSelect, StyledOption } from '@/styles/Combobox'
import { useField } from 'formik'

export const Combobox = ({
  role,
  options,
  placeholder,
  primary,
  onChange,
  name,
  isFormik,
}: ComboboxProps) => {
  const [field, meta] = useField(name)

  const renderSelection = () => {
    if (isFormik) {
      return (
        <StyledSelect primary={primary} {...field} {...meta} role={role}>
          <StyledOption selected value="" name={`${name}.defaultValue`}>
            {placeholder}
          </StyledOption>
          {renderOptions()}
        </StyledSelect>
      )
    }

    return (
      <StyledSelect onChange={onChange} primary={primary}>
        <StyledOption selected value="" name={`${name}.defaultValue`}>
          {placeholder}
        </StyledOption>
        {renderOptions()}
      </StyledSelect>
    )
  }

  const renderOptions = () => {
    return (
      options?.length &&
      options.map(({ value }) => {
        return (
          <StyledOption name={`${name}.${value}`} key={value} value={value}>
            {value}
          </StyledOption>
        )
      })
    )
  }

  return <DropdownWrapper role="combobox">{renderSelection()}</DropdownWrapper>
}
