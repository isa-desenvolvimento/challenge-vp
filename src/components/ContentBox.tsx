import { StyleContainer } from '@/styles/ContentBox'
import { ChildrenProps } from '@/types/global'

export const ContentBox = ({ children }: ChildrenProps) => {
  return <StyleContainer role="contentBox"> {children}</StyleContainer>
}
