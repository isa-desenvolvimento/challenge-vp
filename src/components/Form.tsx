import { StyleFormContainer } from '@/styles/Form'
import { ItemProps } from '@/types/global'
import { Formik } from 'formik'
import React, { EventHandler } from 'react'
import { InputComponent } from '@/components/InputComponent'

export const Form = () => {
  const onSubmit = (data: ItemProps) => {
    console.log(data)
  }

  const renderForm = () => {
    return (
      <Formik
        onSubmit={(values) => onSubmit(values)}
        initialProps={{
          id: '',
          title: '',
          description: '',
          date: '',
          amount: '',
          status: '',
        }}
      >
        {(handleSubmit) => (
          <form onSubmit={handleSubmit}>
            <InputComponent placeholder="R$ 0.0" primary name="title" />
            <InputComponent placeholder="R$ 0.0" primary name="description" />
            <InputComponent placeholder="R$ 0.0" primary name="date" />
            <InputComponent placeholder="R$ 0.0" primary name="amount" />
            <InputComponent placeholder="R$ 0.0" primary name="status" />
          </form>
        )}
      </Formik>
    )
  }

  return <StyleFormContainer>{renderForm()}</StyleFormContainer>
}
