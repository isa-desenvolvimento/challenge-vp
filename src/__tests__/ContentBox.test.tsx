import { renderWithProvider } from '../utils/helpers'
import { screen } from '@testing-library/react'

import { ContentBox } from '@/components/ContentBox'

describe('<ContentBox />', () => {
  beforeEach(() => renderWithProvider(<ContentBox>Test ContentBox</ContentBox>))

  it('should render text ContentBox', () => {
    expect(screen.getByRole(/contentBox/i).textContent).toEqual(
      ' Test ContentBox'
    )
  })

  it('should render the styled-components ContentBox', () => {
    expect(screen.getByRole(/contentBox/i)).toHaveStyle({
      background: 'rgb(239, 240, 246)',
      'grid-area': 'contentBox',
      overflow: 'scroll',
      width: '100%',
      height: '100%',
    })
  })

  it('should render snapshot', () => {
    expect(screen.getByRole(/contentBox/i)).toMatchSnapshot()
  })
})
