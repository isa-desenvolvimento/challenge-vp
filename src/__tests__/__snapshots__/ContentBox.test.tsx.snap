// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ContentBox /> should render snapshot 1`] = `
.c0 {
  grid-area: contentBox;
  overflow: scroll;
  width: 100%;
  height: 100%;
  background: #eff0f6;
}

<main
  class="c0"
  role="contentBox"
>
   
  Test ContentBox
</main>
`;
